docs/conf.py

```diff
@@ -168,6 +168,25 @@
     ],
 }
 
+# Version swticher configuration
+
+# Check if building on readthedocs
+version_match = os.environ.get("READTHEDOCS_VERSION")
+# Check the version type on readthedocs
+version_type = os.environ.get("READTHEDOCS_VERSION_TYPE")
+
+# We want to use the local versions.json if building locally and on PRs (in which case
+# the version type is "external"), and we set the versions accordingly
+
+if version_match is None or version_type == "external":
+    # Use local versions.json if not building on readthedocs
+    # or if building on readthedocs but on PRs
+    html_theme_options["switcher"]["json_url"] = "_static/versions.json"
+
+if version_type == "external":
+    # Set the version to latest on readthedocs PRs
+    html_theme_options["switcher"]["version_match"] = "latest"
```

Comment:
This will only set the `html_theme_options["switcher"]["version_match"]`, it won't work for `stable` nor `latest`. Currently, they'll get `version = pybamm.__version__` (defined in line 33) which is not the version you want to have matched to them IUUC. In addition to these checks to cover local and PR builds, I think you should also have specific checks for stable and latest. They do have independent builds and they appear on the versions.json file, so their builds should have the right value of the version match. Otherwise, users will click on latest and go to `.../en/latest/...` docs, but the "folded" version switcher won't show latest while browsing the docs.

Reply:
Thanks for the review! As noted previously in our conversation—you mentioned that the version switcher is still not going to work for previous documentation releases, so we are deciding to remove the dropdown altogether in order to rely on the server-side readthedocs one. That is, unless there exists a workaround for not having to deal with patch releases with the latest `versions.json` set?